### PR TITLE
KEYCLOAK-17784: URL encode Keycloak's remember-me cookie to allow non-ascii usernames.

### PR DESCRIPTION
**Pull request by bastianccm**
International users using non-ascii symbols such as the german `äöü`
will make Keycloak set the KEYCLOAK_REMEMBER_ME cookie without URL
encoding. This will trigger an java.lang.IllegalArgumentException:
UT000173 exception in undertow's cookie parser which does not
allow non-ascii characters.

Co-authored-by: Fabian Freyer <mail@fabianfreyer.de>

### CODE REVIEW

**Comment by CarlosTenorio** [comment]
Some workarround meanwhile? :(

**Review by mabartos — COMMENTED** [review]
Thanks for contribution! LGTM; I've just started UI tests execution in the test pipeline in order to ensure correctness of this solution. If these tests pass, I'll approve it.

File: testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/realm/LoginSettingsTest.java [hunk 1]
```diff
@@ -232,6 +232,44 @@ public void rememberMe() {
         
     }
     
+    @Test
+    public void rememberMeWithUtf8Username() {
+        log.info("creating username with non-ASCII username");
+        UserRepresentation utf8TestUser = createUserRepresentation("täst", "test-utf8@email.test", "utf8", "test", true);
+        setPasswordFor(utf8TestUser, PASSWORD);
+
```
Comment (mabartos) [inline testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/realm/LoginSettingsTest.java:240]: _UserRepresentation_ is basically DTO, but the user is not created on the server.
Please import this:
```
import static org.junit.Assert.assertNotNull;
import static org.keycloak.testsuite.admin.ApiUtil.createUserWithAdminClient;
```
Create user on the server.
```suggestion
String id = createUserWithAdminClient(testRealmResource(), utf8TestUser);
assertNotNull(id);
```

**Review by mabartos — CHANGES_REQUESTED** [review]
@bastianccm Unfortunately, the UI tests failed due to inaccuracy in the new test case. You can execute the UI tests on your own machine. It's necessary to run it with supported browser, please take a look at [this](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/HOW-TO-RUN.md#different-browsers). 
f.e.
`mvn -f testsuite/integration-arquillian/tests/other/console clean install -Dtest=LoginSettingsTest#rememberMeWithUtf8Username -Dbrowser=chrome`

The test fix [here](https://github.com/mabartos/keycloak/commit/747c02b3e95dbf1cd607e4f6e5e11baf6be534b6).

**Comment by bastianccm** [comment]
@mabartos thank you so much for the fix!

**Review by mabartos — APPROVED** [review]
@bastianccm Great job! The UI tests passed. (https://keycloak-jenkins.com/job/universal-test-pipeline-server/375/)

@mposolda Could you please take a look at it?

**Comment by mposolda** [comment]
@bastianccm @mabartos Question: Will it work if you have KEYCLOAK_REMEMBERME cookie set from previous version? I mean scenario like this:
- Keycloak 13 started
- User "john" authenticated with rememberme. KEYCLOAK_REMEMBERME cookie created (in the old format)
- Keycloak 13 stopped and new Keycloak with this change started
- What happen when user "john" opens his browser? IMO it is fine if user needs to re-authenticate in this case. But it is not fine if there is some RuntimeException thrown to the user and there is some manual intervention needed for him (like manually delete cookie)

**Comment by mabartos** [comment]
@mposolda Thanks for your comment. It works with previous version, user has to re-authenticate and there's no need for some manual intervention.

**Comment by mposolda** [comment]
@mabartos Thanks for the doublecheck!

**Review by mposolda — APPROVED** [review]

**Pull request merged**